File: SemanticSearch/Ruby/Cli/semantic_search.rb
def list_collections
  driver = ENV.fetch('VECTOR_DRIVER', 'chromadb')
  puts "Collections: (driver: #{driver})"
  puts "  - documents"
end

def search(query)
  puts "Searching for: #{query}"
  puts "  No results found (demo mode)"
end

def delete_collection(name)
  puts "Collection '#{name}' deleted"
end

puts "Semantic Search CLI"
puts "1. List collections"
puts "2. Search documents"
puts "3. Delete collection"
puts "4. Exit"
print "Choose an option: "
choice = gets.to_s.strip

case choice
when '1'
  list_collections
when '2'
  print "Search query: "
  q = gets.to_s.strip
  search(q)
when '3'
  print "Collection name: "
  name = gets.to_s.strip
  delete_collection(name)
when '4'
  puts "Goodbye!"
end

File: SemanticSearch/PHP/Cli/semantic_search.php
<?php

function listCollections(): array {
    $driver = getenv('VECTOR_DRIVER') ?: 'chromadb';
    $adapter = match ($driver) {
        'pinecone' => new PineconeAdapter(),
        'pgvector' => new PgVectorAdapter(),
        default => new ChromaDBAdapter(),
    };
    $adapter->connect();
    return $adapter->listCollections();
}

function search(string $query): array {
    $driver = getenv('VECTOR_DRIVER') ?: 'chromadb';
    $adapter = match ($driver) {
        'pinecone' => new PineconeAdapter(),
        'pgvector' => new PgVectorAdapter(),
        default => new ChromaDBAdapter(),
    };
    $adapter->connect();
    $dimension = (int)(getenv('VECTOR_DIMENSION') ?: 1536);
    return $adapter->search(array_fill(0, $dimension, 0.0), 5);
}

function deleteCollection(string $name): void {
    $driver = getenv('VECTOR_DRIVER') ?: 'chromadb';
    $adapter = match ($driver) {
        'pinecone' => new PineconeAdapter(),
        'pgvector' => new PgVectorAdapter(),
        default => new ChromaDBAdapter(),
    };
    $adapter->connect();
    $adapter->deleteCollection($name);
}

echo "Semantic Search CLI\n";
echo "1. List collections\n";
echo "2. Search documents\n";
echo "3. Delete collection\n";
echo "4. Exit\n";
echo "Choose an option: ";
$choice = trim(fgets(STDIN));

switch ($choice) {
    case '1':
        $collections = listCollections();
        foreach ($collections as $c) echo "  - $c\n";
        break;
    case '2':
        echo "Search query: ";
        $q = trim(fgets(STDIN));
        $results = search($q);
        foreach ($results as $r) echo "  [{$r['distance']}] " . substr($r['document'], 0, 100) . "\n";
        break;
    case '3':
        echo "Collection name: ";
        $name = trim(fgets(STDIN));
        deleteCollection($name);
        echo "Collection '$name' deleted\n";
        break;
    case '4':
        echo "Goodbye!\n";
        break;
}